### DISCUSSION TITLE
Want to develop custom cards

### DISCUSSION THREAD

**Discussion by spencersmb** [UI-Lovelace-Minimalist/UI#1065] [Q&A]
I love the way lovelace-ui looks but I'm struggling with building my own cards from scratch. I would love a tutorial on how to develop cards. I come from a web background and for example having to restart HA every time I make a change to see if it worked seems counter intuitive to a speedy workflow. Any tips? Tutorials? Things that can help me build my own components and cards?

Thanks!

**Comment by basbruss** [comment]
Hi

You don't need to restart HA every time.
<img width="957" alt="image" src="https://user-images.githubusercontent.com/68892092/197324144-33d6f4d7-96a1-49ab-828c-e46d78b63e52.png">
or you can use this:
<img width="682" alt="image" src="https://user-images.githubusercontent.com/68892092/197324240-e936b947-ab61-4124-afd5-e6cd8508861f.png">


This pages can be handy:
https://ui-lovelace-minimalist.github.io/UI/setup/custom_cards/
https://ui-lovelace-minimalist.github.io/UI/development/card_style_guide/
https://ui-lovelace-minimalist.github.io/UI/development/custom_cards/

**Comment by spencersmb** [comment]
HI basbruss,

Thank you for that information. I'm currently working on a replacement for my theater remote using HA:
<img width="804" alt="image" src="https://user-images.githubusercontent.com/9687849/197348441-8735f4ec-f2fa-44af-9207-08297b217cfc.png">

I have two question so far after digging through everything last night and today:

Grid layouts:
Is there any documentation on how implementing css grid works?
I would love to have greater control over the grid and how things are organized from different screen sizes.

Colors:
Currently in my layout I only have blue colored buttons, is there a way to change the colors without coding a completely new component just to change the color?

Thanks!
